packages/server/src/handlers/GetHandler.ts

```diff
@@ -1,14 +1,57 @@
 import stream from 'node:stream'
 
 import {BaseHandler} from './BaseHandler'
-import {ERRORS} from '@tus/utils'
+import {ERRORS, Upload} from '@tus/utils'
 
 import type http from 'node:http'
 import type {RouteHandler} from '../types'
 
 export class GetHandler extends BaseHandler {
   paths: Map<string, RouteHandler> = new Map()
 
+  /**
+   * reMimeType is a RegExp for check mime-type form compliance with RFC1341
+   * for support mime-type and extra parameters, for example:
+   *
+   * ```
+   * text/plain; charset=utf-8
+   * ```
+   *
+   * See: https://datatracker.ietf.org/doc/html/rfc1341 (Page 6)
+   */
+  reMimeType =
+    /^(?:application|audio|example|font|haptics|image|message|model|multipart|text|video|x-(?:[0-9A-Za-z!#$%&'*+.^_`|~-]+))\/([0-9A-Za-z!#$%&'*+.^_`|~-]+)((?:[ 	]*;[ 	]*[0-9A-Za-z!#$%&'*+.^_`|~-]+=(?:[0-9A-Za-z!#$%&'*+.^_`|~-]+|"(?:[^"\\]|\.)*"))*)$/
```

Comment:
Is it worth it to support parameters as well? We now have a very complex, slow regex with lots of backtracking. If it doesn't add much value I think I prefer to stay with the simple version? What do you think?

Reply:
I understand the caution, this last regex is definitely much more intimidating than the previous one, but it is not significantly slower, less than 2ms on average, and this last one analyzing the list of 2138 mime types registered by the IANA.

And answering your question, I think it is important, especially for text files, to preserve and transmit the information of the character set used, and this can be rendered properly when displayed inline.

If you like, we can change to an intermediate version that does not try to match on the parameters, and only looks at the mime type part, after that "anything can come".

## Benchmarks

### Simple Regex
![Simple Regex](https://github.com/user-attachments/assets/f7f94547-c4b2-4c96-84ef-591a4e36c23a)

### Allow Parameters and RFC compliance Regex
![Allow Parameters Regex](https://github.com/user-attachments/assets/e66726d1-1b60-4a1d-8012-cf15433664ae)

### Non Parameters match Regex
![Intermediate Regex](https://github.com/user-attachments/assets/4dc50f55-d898-41a9-9b91-7b9fdf6cd3ed)

Reply:
Parameters in `Content-Type` are also relevant for some media files. Although I am not sure if necessary, they can contain information about the used codecs in audio/video files (see https://github.com/tus/tusd/issues/1194).

Tusd currently does not support parameters, but I plan an changing this. Its implementation currently also uses a regular expression for checking the media type's validity, but my preferred solution is to replace it with [Go's builtin media type parser](https://pkg.go.dev/mime#ParseMediaType). I'm not sure if such a method is easily available to you, but maybe it provides another perspective on this topic.

Reply:
@jesusgoku wow thanks for taking the time to actually test it. Sounds good to me.